src/confluent_kafka/src/confluent_kafka.h

```diff
@@ -275,8 +275,66 @@ int  Handle_traverse (Handle *h, visitproc visit, void *arg);
 typedef struct {
 	PyThreadState *thread_state;
 	int crashed;   /* Callback crashed */
+	PyObject *exception_type;    /* Stored exception type */
+	PyObject *exception_value;   /* Stored exception value */
+	PyObject *exception_traceback; /* Stored exception traceback */
 } CallState;
 
+/**
+ * @brief Compatibility layer for Python exception handling API changes.
+ * PyErr_Fetch/PyErr_Restore were deprecated in Python 3.12 in favor of 
+ * PyErr_GetRaisedException/PyErr_SetRaisedException.
+ */
+
+static inline void
+cfl_exception_fetch(PyObject **exc_type, PyObject **exc_value, PyObject **exc_traceback) {
+#if PY_VERSION_HEX >= 0x030c0000
+    /* Python 3.12+ - use new API */
+    PyObject *exc = PyErr_GetRaisedException();
+    if (exc) {
+        *exc_type = (PyObject *)Py_TYPE(exc);
+        Py_INCREF(*exc_type);
+        *exc_value = exc;
+        *exc_traceback = PyException_GetTraceback(exc);
```

Comment:
We don't need `exc_type` and `exc_traceback` in the new C APIs. Check if we can remove these safely. This removal will simplify ref count handling as well.

Check `CallState_end` as well. `cs->exception_type` usage is added there in the PR.

```
if (cs->crashed) {
		/* Restore the saved exception if we have one */
		if (cs->exception_type) {
			CallState_restore_exception(cs);
		}
		return 0;
	}
```